### ISSUE TITLE
google photos slide show not working (only showing one photo from the google album).

### ISSUE THREAD

**Issue by janvda** [balena-io-projects/balena-dash#56]
Hi,

I am trying the get a slide show of one my shared google albums but it is only showing one photo from the album.

FYI here below my log file.
```
20.11.19 17:37:45 (+0100) Restarting service 'photos sha256:dcee19a56e592e32ce93de7a975b6f6abf1f55808253df949de95731c523db32'
20.11.19 17:37:48 (+0100)  photos  
20.11.19 17:37:48 (+0100)  photos  > album-slideshow@1.0.0 start /
20.11.19 17:37:48 (+0100)  photos  > node server.js
20.11.19 17:37:48 (+0100)  photos  
20.11.19 17:37:50 (+0100)  photos  📷 - Fetching images...
20.11.19 17:37:50 (+0100)  photos  📷 - Starting a google photos slideshow.
20.11.19 17:37:50 (+0100)  photos  ✨ - Magic happens on port 8888
20.11.19 17:37:52 (+0100)  photos  Found 1 images.
20.11.19 17:37:52 (+0100)  photos  🔀 - Shuffling images OFF 🔀
```

**Comment by janvda** [comment]
I think that the following part of the `server.js` code of the `photo` service is responsible that only one image is found:

```
    // Parse photos
    const rx = /\["(https:\/\/[^\.]+.googleusercontent\.com\/[^"]+)",([0-9]+),([0-9]+),/;
    const extractPhotos = data =>
      data
        .match(new RegExp(rx, "g"))
        .map(m => m.match(rx))
        .map(p => {
          const width = +p[2];
          const height = +p[3];
          const url = `${p[1]}=w${width}-h${height}-no`;
          return `${p[1]}=w1000`; // Change here to increase/decrease image width-size
        });

    request(
      {
        uri: albumURL,
        method: "GET",
        timeout: 10000
      },
      function(error, response, body) {
        if (!error && response.statusCode == 200) {
          // Fetch and remove duplicate images
          photos = extractPhotos(body);
          photos = [...new Set(photos)]; // save only unique values

          console.log("Found " + photos.length + " images.");
          setImagesArray(photos);
        }
```

**Comment by janvda** [comment]
FYI why is it not recognizing the following 2 URLs in the body:

```

}});</script><script nonce="Jo7wKDvoLT1B8ZUYoI+riw">AF_initDataCallback({key: 'ds:1', isError:  false , hash: '3', data:function(){return [null,[["AF1QipN71Detc9AP5-WO0V_tVEZxcUnTr9QSL6a8V5i9",["https://lh3.googleusercontent.com/kpnpXGBLkRplXRHVR2iuGBN-jHqy0h3_zgT1noUxonmYC2RupdNCK2CqxOzP7AxwCgkmrriEeJ-tvxxGIyAAAXNH8kk9WK_jCT6KXTQBsRCW9asvG7_O1_r_-ytRrsuGh0mL-zm8PGUuXXm36wAx5yuJ-uuKph1ixZ9biQqDqxTn1VV1CGaZb8VGLT0JupOb-iJyYpPVVH2p558YRQeg3Mt_jM5DRLsBB0OCCM868WKe4hT-WC0KmK5yuH1liAhijhwSlBhramgYfuwVbRQ6jy6OCal61UP_UZ32wM89WYs0AAzRCLppTImgyfna4sopkOuArBfdz0iTirTSoTbZbvNRVWviRGRRZxYwR56WIkCeusfavUdgb_ffNBu2Br8CNIq6DDGKlCUU-7EsTKmz_OHLsyQOeyOlaX_CEzmY0roivPGHJsGoaeQ_4GGWlh3B69FWICe_ngO_VglKROdGmXSts3OQDMsOkT11vW7GnxthZ8jxdAGZZFRInhZcvSP1crr4nzqpSGR45XOsPtRDlr_Qyml5vjM-htq9OdhBPOYgEJpxUaB7wOXik-548dDVVGKiTL_ZTE0RMDzP6l7K0_xzH53fYqpNdSbHa1A57_Bv4aLyek_lL3RbBfqKMxfxM5osyiimk2bP4Lw965E-v75fUbtExFfX8GLDSyKros8F_BEHKy8LuQ-7275rXFdKRaQu5NcuwazKpOK2CWhNC3p_PbO6jzbGFiN78diOB7hBJVo",1024,768]
--
  | ,1153347668000,"HaaqqAtqjKRKhWfHn4xxPCfHQX0",0,1218719434000,["AF1QipN2snJuP5KSMKm6cx-q1dW0A2Ik"]
  | ,[[2]
  | ,[9]
  | ,[5]
  | ,[31,false,true]
  | ,[36,false,true]
  | ,[21]
  | ,[20]
  | ,[10]
  | ,[19]
  | ,[22]
  | ]
  | ,2,null,null,null,{"101428965":[0.0,"vd2btbut"]
  | }]
  | ,["AF1QipO2N982Dzt_XrXNfWRaboSTiOAq57AfvHbTdLxf",["https://lh3.googleusercontent.com/rTXxwFN4GyFuxXDJ10Krn1Wd-7bLd5JbYWepdSeGNWKjE8MRApG7sWVeZiXg2hD9uSYsr8y9yvAL93XfT4zjdYbAYC-Sj4uo6a7ZfmbyFscIKS7aA9pffTMId-PR3T-XPJSR9HFz2ZwQ5bI4y2LgmPIUlmkK7abLSx2khSW5jBmrxtMn90ndGpvie3lDa-hYyn8ErFfmlKFpjhnfWQP7hTjLV3NKRO9e5dGUTQ7pyIjiXeZwV-Nmjxdss7UpGG7d040g5_duOR6K8hkiqoxbuMQ7RRdK_SbeT-9BJUfh1Dk7-QB240EGaA7l_gyQewz2eNLlmdUbJ_PgbdhBdSRxI7VRU6Cqa987NOE4RVXXTECnF5EgfKzivvN4EHMEN581DguY3K7YGvMqrSJP51dL14KlGHCjLapzz-WCzfHiKVFZA3CCN3W6I-n6b4eYKKlluqviz0HyVMc885Aogr8DPzbuAiTrm98_49rU0-fe_Wkv5I7_6Ykfjcy0Hnzhxj_AqG7nsBkt-8WF2bGuqVnVd6gT8kjd-vdNzzJaa7saS8m5VKdREhW_K0Ullet_AhuLzmlhU349DoBQIpSJoc4C91WTxAiurd3RubHeHmSo06iqOqVJ76j0dkTYuWTeT7NVtEOWujX2GTyWsdujaeMUTLupLC5ZM72xZZ93x6KmLJEfe00ttjoMWFTS0NuyADN7QFtu2vgCvubTlSThcrHXSB2-v_bXmmzEfavXc9lvmO30Twg",1024,768]
  | ,1153347695000,"SuRkanWSIFUC_ii8zX0vEKLkvoQ",0,1218719436000,["AF1QipN2snJuP5KSMKm6cx-q1dW0A2Ik"]
```

**Comment by janvda** [comment]
I have found the issues.
The above 2 URLs are not ending with `,` after the width and height but with a `]`.

So this means that the regular expression is not recognizing these 2 cases.
I have fixed it by changing the regular expression from
```
const rx = /\["(https:\/\/[^\.]+.googleusercontent\.com\/[^"]+)",([0-9]+),([0-9]+),/;
```
into (so it accepts both a `,` and a `]` after the height).
```
const rx = /\["(https:\/\/[^\.]+.googleusercontent\.com\/[^"]+)",([0-9]+),([0-9]+)[,\]]/;
```

**Comment by janvda** [comment]
Fixed by pull request #57
